modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinCacheToJdbcDataTypesCoverageTest.java

```diff
@@ -0,0 +1,620 @@
+
+        assert stmt != null;
+        assert !stmt.isClosed();
+    }
+
+    /**
+     * Remove record form cache via cache API, and verify that data was successfully removed with sql jdbc query.
+     *
+     * @param tblName Table name.
+     * @param cache Ignite cache.
+     * @param originalKey Original Key.
+     * @throws SQLException If Failed.
+     */
+    private void checkDelete(String tblName, IgniteCache<Object, Object> cache,
+        Object originalKey) throws IgniteCheckedException, SQLException {
+        // Delete from cache.
+        cache.remove(originalKey);
+
+        try {
+            if (writeSyncMode == CacheWriteSynchronizationMode.FULL_ASYNC &&
+                !waitForCondition(new GridAbsPredicateX() {
+                                      @Override public boolean applyx() throws IgniteCheckedException {
+                                          try {
+                                              return !stmt.executeQuery("SELECT * FROM " + tblName).next();
+                                          }
+                                          catch (SQLException e) {
+                                              throw new IgniteCheckedException(e);
+                                          }
+                                      }
+                                  },
+                    TIMEOUT_FOR_KEY_RETRIEVAL_IN_FULL_ASYNC_MODE))
+                fail("Deleted data are still retrievable via SELECT.");
+        }
+        catch (GridClosureException e) {
+            throw (SQLException)e.getCause().getCause();
+        }
+
+        ResultSet rs = stmt.executeQuery("SELECT * FROM " + tblName);
+
+        assertNotNull(rs);
+
+        assertFalse("Unexpected rows count.", rs.next());
+    }
+
+    /**
+     *
+     * @param converterToSqlExpVal Fгnction that converts expected key/value in conformity with a format of jdbc
+     *  return key/value.
+     * @param equalsProcessor Equals processor that process equality check of expected and got keys/values.
+     * @param dataType Data type.
+     * @param originalKey Original key.
+     * @param originalVal Original value.
+     * @param selectQry Select query to execute.
+     * @throws IgniteCheckedException If failed.
+     * @throws SQLException If failed.
+     */
+    @SuppressWarnings("unchecked")
+    private void checkQuery(Function converterToSqlExpVal, BiFunction<Object, Object, Boolean> equalsProcessor,
+        Class<?> dataType, Object originalKey, Object originalVal, String selectQry)
+        throws IgniteCheckedException, SQLException {
+        try {
+            if (writeSyncMode == CacheWriteSynchronizationMode.FULL_ASYNC &&
+                !waitForCondition(new GridAbsPredicateX() {
+                                      @Override public boolean applyx() throws IgniteCheckedException {
+                                          try {
+                                              return stmt.executeQuery(selectQry).next();
```

Comment:
Cursor is not closed here.

Reply:
Seems that it's not a problem cause they are closed either inside stmt.execute() or within afterTest()